libsel4vmmplatsupport/src/drivers/pci_helper.c

```diff
@@ -255,19 +242,6 @@ vmm_pci_entry_t vmm_pci_create_bar_emulation(vmm_pci_entry_t existing, int num_b
     };
 }
 
-vmm_pci_entry_t vmm_pci_create_passthrough_bar_emulation(vmm_pci_entry_t existing, int num_bars, vmm_pci_bar_t *bars)
-{
-    pci_bar_emulation_t *bar_emul = calloc(1, sizeof(*bar_emul));
-    assert(bar_emul);
-    memcpy(bar_emul->bars, bars, sizeof(vmm_pci_bar_t) * num_bars);
-    bar_emul->passthrough = existing;
-    bar_emul->num_bars = num_bars;
-    memset(bar_emul->bar_writes, 0, sizeof(bar_emul->bar_writes));
-    return (vmm_pci_entry_t) {
-        .cookie = bar_emul, .ioread = pci_bar_passthrough_emul_read, .iowrite = pci_bar_passthrough_emul_write
-    };
-}
-
```

Comment:
Can you please move the removal of this function, it's implementation and it's documentation to a separate commit. Also in the commit message can you explain why it should be removed. (It appears to me that it's a no-op as it could be implemented by just returning `existing`).

Reply:
Done.

Reply:
I think you can find the explanation in the commit itself, just for sake of convenience that kind of emulation existed only to prevent doing double emulation on BAR registers; the fault handler did that anyway, so we had to avoid doing that in the emulation code before also.